miasm2/core/asmblock.py

```diff
@@ -359,132 +323,209 @@ class AsmSymbolPool(object):
     """
 
     def __init__(self):
-        self._labels = set()
-        self._name2label = {}
-        self._offset2label = {}
-        self._label_num = 0
+        self._loc_keys = set()
+
+        self._loc_key_to_offset = {}
+        self._loc_key_to_name = {}
+
+        self._name_to_loc_key = {}
+        self._offset_to_loc_key = {}
+
+        self._loc_key_num = 0
 
-    def add_label(self, name, offset=None):
+    def loc_key_to_offset(self, loc_key):
         """
-        Create and add a label to the symbol_pool
-        @name: label's name
-        @offset: (optional) label's offset
+        Return offset of @loc_key, None else.
+        @loc_key: LocKey instance
         """
-        label = AsmLabel(name, offset)
+        return self._loc_key_to_offset.get(loc_key, None)
+
+    def loc_key_to_name(self, loc_key):
+        """
+        Return name of @loc_key.
```

Comment:
Comment of `loc_key_to_XXX` and `XXX_to_loc_key` should have a common form, ie. specifying the conversion and `None` default case.
The class docstring could have in its doc something listing the common interfaces idea.

This can also be reported to the patch introducing the future LocationDB

Reply:
See with @commial: not a bug as name is mandatory